Code/GraphMol/Kekulize.cpp

```diff
@@ -460,7 +460,12 @@ void kekulizeFused(RWMol &mol, const VECT_INT_VECT &arings,
     // number of backTracks) and we still need to backtrack
     // can't kekulize this thing
     std::ostringstream errout;
-    errout << "Can't kekulize mol " << std::endl;
+    errout << "Can't kekulize mol.";
+    errout << "  Atoms remaining:";
+    for (unsigned int i = 0; i < nats; ++i) {
+      if (dBndCands[i]) errout << " " << i;
```

Comment:
Perhaps a strict or pedantic mode that captures this info in a sanitizer exception might work.  The problem is that you don't really want this for mol suppliers,
Although I can imagine it might be useful for logging...

----
Brian Kelley

> On May 20, 2016, at 7:36 AM, Greg Landrum <notifications@github.com> wrote:
> 
> In Code/GraphMol/Kekulize.cpp:
> 
> > @@ -460,7 +460,12 @@ void kekulizeFused(RWMol &mol, const VECT_INT_VECT &arings,
> >      // number of backTracks) and we still need to backtrack
> >      // can't kekulize this thing
> >      std::ostringstream errout;
> > -    errout << "Can't kekulize mol " << std::endl;
> > +    errout << "Can't kekulize mol.";
> > +    errout << "  Atoms remaining:";
> > +    for (unsigned int i = 0; i < nats; ++i) {
> > +      if (dBndCands[i]) errout << " " << i;
> yes. They are the atoms that were on the list to be kekulized but that were not.
> Doing something with the _supplementalSmilesLabel is an interesting idea, but it would be difficult to do much with since you typically don't actually have access to the molecule when kekulization fails. Users would have to detect the failure, reconstruct the molecule without sanitization, sanitize and trap the exception, and then generate the SMILES in order to be able to see that info. It's a lot of work.
> The first order solution - adding SMILES reporting here - can't be used because we don't want to add a dependency on SmilesParse to the GraphMol lib.
> 
> Any other thoughts on that?
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
>